### PR TITLE
Add mx.pwmc.Svgvi application

### PR DESCRIPTION
**Pull request by pwmcmx**
New application request. SvgVi is an effort to provide precision editable SVG files with preview while it is edited.

### CODE REVIEW

File: mx.pwmc.Svgvi.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options": {
```
Comment (nedrichards) [inline mx.pwmc.Svgvi.json]: these are all defaults in this version and you should remove them.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi
testbot, build mx.pwmc.Svgvi

File: mx.pwmc.Svgvi.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
+{
+    "app-id": "mx.pwmc.Svgvi",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.30",
+    "sdk": "org.gnome.Sdk",
+    "writable-sdk": true,
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: This shouldn't be needed. What error did you fix with it?

File: mx.pwmc.Svgvi.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    "command": "mx.pwmc.Svgvi",
+    "finish-args": [
+        "--share=network",
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: What does it use networking for?

File: mx.pwmc.Svgvi.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home",
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: Could you limit it to `xdg-pics`?

File: mx.pwmc.Svgvi.json [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                        "type": "archive",
+                        "url": "https://download.gnome.org/sources/gxml/0.16/gxml-0.16.3.tar.xz",
+                        "sha256": "520d4d779b1d31591762b2a98f84072531b9e17ac401df9668493e189eafc6ba"
+                        }
+                ]
+        },
+        {
+                "name": "gresg",
+                "buildsystem": "meson",
+                "builddir": true,
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: It's already default for meson.

Comment (barthalion) [reply mx.pwmc.Svgvi.json]: (also applies below)

File: mx.pwmc.Svgvi.json [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                                "type": "archive",
+                                "url": "https://download.gnome.org/sources/gtksourceview/3.24/gtksourceview-3.24.9.tar.xz",
+                                "sha256": "699d76a453e6a3d3331906346e3dbfa25f2cbc9ec090e46635e9c6bb595e07c2"
+                        }
+                ]
+        },
+        {
+                "name": "svgvi",
+                "buildsystem": "meson",
+                "config-opts": [ "--libdir=lib" ],
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: Isn't it the default as well?

File: mx.pwmc.Svgvi.json (hunk 1 above)
Comment (pwmcmx) [reply mx.pwmc.Svgvi.json]: Ok. Done.

File: mx.pwmc.Svgvi.json (hunk 2 above)
Comment (pwmcmx) [reply mx.pwmc.Svgvi.json]: This is need by dependencies:
` "writable-sdk": true `

They add GObject Introspection GIR/TYPELIB and without this setting it is not possible and fails.

File: mx.pwmc.Svgvi.json (hunk 3 above)
Comment (pwmcmx) [reply mx.pwmc.Svgvi.json]: Any file on the network can be opened for edition, thanks to GIO.

File: mx.pwmc.Svgvi.json (hunk 4 above)
Comment (pwmcmx) [reply mx.pwmc.Svgvi.json]: No. This is a design tool, so it needs user's home directory access.

File: mx.pwmc.Svgvi.json (hunk 5 above)
Comment (pwmcmx) [reply mx.pwmc.Svgvi.json]: `gresg` still have both meson and Autotools, until the last is removed I need to set the buildsystem to use.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi
testbot, build mx.pwmc.Svgvi

**Comment by pwmcmx** [comment]
I was trying to use librsvg 2.40.20, because the one shipped in FlatHub is buggy around to SVG rendering see at: https://gitlab.gnome.org/GNOME/librsvg/issues/412

But build fails!

So I've removed this module, but the application will not run as good as it should be.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi
testbot, build mx.pwmc.Svgvi

**Comment by pwmcmx** [comment]
Last fails spot a limitation on this build bot, because it doesn't search for meson.build files in order to try to build using meson.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi
testbot, build mx.pwmc.Svgvi

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi

**Comment by barthalion** [comment]
bot, build mx.pwmc.Svgvi

File: mx.pwmc.Svgvi.json (hunk 2 above)
Comment (barthalion) [reply mx.pwmc.Svgvi.json]: Sounds like it just installs files at wrong location. Which dependencies do so?

```
"build-options": {
        "env": {
            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_GIRDIR": "/app/share/gir-1.0",
            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_TYPELIBDIR": "/app/lib/girepository-1.0"
        }
    },
```

is usually enough to make it work.

File: mx.pwmc.Svgvi.json (hunk 5 above)
Comment (barthalion) [reply mx.pwmc.Svgvi.json]: What I mean is that `"buildsystem": "meson"` turns `"builddir"` on.

File: mx.pwmc.Svgvi.json [hunk 7]
```diff
@@ -0,0 +1,121 @@
+{
+    "app-id": "mx.pwmc.Svgvi",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.30",
```
Comment (barthalion) [inline mx.pwmc.Svgvi.json]: Would be nice to give 3.32 a shot.

**Comment by barthalion** [comment]
Builds fails on:

```

**Comment by barthalion** [comment]
bot, build mx.pwmc.Svgvi

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi

**Comment by nedrichards** [comment]
fwiw the URL in the app is broken, it seemingly should be: https://www.pwmc.mx/en_US/page/svgvi-1 but other than that, it works! Renders SVG tiger pretty well ;-)

File: mx.pwmc.Svgvi.json [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.32",
+  "sdk": "org.gnome.Sdk",
+  "command": "mx.pwmc.Svgvi",
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--filesystem=home",
```
Comment (nedrichards) [inline mx.pwmc.Svgvi.json]: It looks like the only filesystem interaction is pretty simple though file open and save. Can you not switch the FileChooser calls to FileChooserNative and automatically get Portals support and remove the filesystem entirely? https://gitlab.com/search?utf8=%E2%9C%93&search=filechooser&group_id=&project_id=10414336&search_code=true&repository_ref=master doesn't look like you're that custom and https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html should support that.

Comment (esodan) [reply mx.pwmc.Svgvi.json]: I interpreted your request to use FileChooserNative, so I've switched to it.

About file system, are you suggesting to remove `"--filesystem=home" `, if that the case how do the application will get a file in the local or a network file system?

Comment (esodan) [reply mx.pwmc.Svgvi.json]: I've tested locally and removing --filesystem=home while using Native file chooser, have no impact. So I've switched to them.
Updating the manifest.

**Comment by nedrichards** [comment]
Oh, also:
> (mx.pwmc.Svgvi:3): dconf-WARNING **: 22:56:03.878: failed to commit changes to dconf: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown

looks like you need to add the dconf permissions.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi

**Comment by esodan** [comment]
Last commit, hope it fixes dconf access and get it approved.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi

File: mx.pwmc.Svgvi.json [hunk 9]
```diff
@@ -10,7 +10,9 @@
     "--socket=x11",
     "--socket=wayland",
     "--filesystem=xdg-run/dconf",
-    "--filesystem=~/.config/dconf:ro"
+    "--filesystem=~/.config/dconf:ro",
+    "--talk-name=ca.desrt.dconf",
+    "-env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (nedrichards) [inline mx.pwmc.Svgvi.json]: you're missing a - here.

**Comment by nedrichards** [comment]
bot, build mx.pwmc.Svgvi

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/mx.pwmc.Svgvi

**Comment by nedrichards** [comment]
Let me know if there's any more users that need to be added to the repo.

**Comment by pwmcmx** [comment]
Many thanks for all your help.

**Pull request closed without merging**